app/src/main/java/com/allan/boardbuddies/repositories/BoardRepository.java

```diff
@@ -0,0 +1,83 @@
+package com.allan.boardbuddies.repositories;
+
+import androidx.lifecycle.LiveData;
+import androidx.lifecycle.MutableLiveData;
+
+import com.allan.boardbuddies.Utilities;
+import com.allan.boardbuddies.models.Board;
+import com.allan.boardbuddies.models.Note;
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+
+import java.io.File;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Comparator;
+
+public class BoardRepository {
+    private static volatile BoardRepository boardRepository;
+
```

Comment:
interesting...is this how it's done in java? to maintain a single instance for the entire application lifecycle?

Reply:
Idk I just ended up using the singleton architecture. If you have a better recommendation we can talk